includes/functions/functions_templates.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * @copyright Copyright 2003-2020 Zen Cart Development Team
+ * @license https://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version $Id: New in v1.5.8 $
+ */
+if (!defined('IS_ADMIN_FLAG')) {
+    die('Illegal Access');
+}
+
+/**
+ * Get all template directories found in catalog folder structure
+ *
+ * @return array
+ */
+function zen_get_catalog_template_directories()
+{
+    if (!defined('DIR_FS_CATALOG_TEMPLATES')) {
+        die('Fatal error: DIR_FS_CATALOG_TEMPLATES not defined.');
```

Comment:
How about a `trigger_error` here, and below, prior to the `die` so that there's a record of the backtrace associated with the issue?